src/components/engagement/handlers/apply-finalized-changeset-to-engagement.handler.ts

```diff
@@ -64,37 +64,19 @@ export class ApplyFinalizedChangesetToEngagement
             ])
             .apply((q) =>
               status === ProjectChangeRequestStatus.Approved
-                ? q.setValues({
-                    'engagementRel.active': true,
-                  })
+                ? q
+                    .setValues({
+                      'engagementRel.active': true,
+                    })
+                    .with('node, changeset')
+                    .apply(commitChangesetProps())
```

Comment:
Not sure why it does not apply to the pending changeset values.
```
MATCH (project:Project)-[:changeset { active: true }]->(changeset:Changeset { id: 'RquE9ZVBog7' })
    CALL {
      WITH project
      MATCH (project)-[engagementRel:engagement { active: true }]->(node:Engagement)
      RETURN node
      UNION
      WITH project, changeset
      MATCH (project)-[engagementRel:engagement { active: false }]->(node:Engagement)<-[changesetRel:changeset { active: true }]-(changeset)
      SET engagementRel.active = true
      WITH node, changeset
      CALL {
        WITH node, changeset
        // For all changed properties of node & changeset
        MATCH (node)-[changedRel { active: false }]->(changedProp:Property)<-[:changeset { active: true }]-(changeset)
        // Apply previous label to active prop
        CALL {
          WITH changeset, node, changedRel
          MATCH (node)-[previouslyActiveRel { active: true }]->(previouslyActiveProp:Property)
          WHERE type(previouslyActiveRel) = type(changedRel)
          CREATE (previouslyActiveProp)<-[:previous { active: true, createdAt: datetime() }]-(changeset)
          RETURN previouslyActiveProp, previouslyActiveRel
        }
        WITH changedRel, previouslyActiveProp, previouslyActiveRel
        // Set changed prop to active
        SET changedRel.active = true
        // Deactivate currently active prop
        SET previouslyActiveRel.active = false
        WITH previouslyActiveProp
        // prefixNodeLabelsWithDeleted(previouslyActiveProp)
        CALL {
          WITH previouslyActiveProp
          WITH previouslyActiveProp, reduce(
            deletedLabels = [], label in labels(previouslyActiveProp) |
              case
                when label starts with "Deleted_" then deletedLabels + label
                else deletedLabels + ("Deleted_" + label)
              end
          ) as deletedLabels
          call apoc.create.removeLabels(previouslyActiveProp, labels(previouslyActiveProp)) yield node as nodeRemoved
          WITH previouslyActiveProp, deletedLabels
          call apoc.create.addLabels(previouslyActiveProp, deletedLabels) yield node as nodeAdded
          RETURN 1
        }
        RETURN 1
      }
      RETURN node
    }
    RETURN node.id as id
```
Can you let me know what I am wrong with the query?

Reply:
BaseNodes created in changesets don't need their props committed as they are [just active](https://github.com/SeedCompany/cord-api-v3/blob/narrative-report-form/src/core/database/database.service.ts#L292-L308).


Is that the problem?

Reply:
Hmm, I don't think so because the [test](https://github.com/SeedCompany/cord-api-v3/blob/develop/test/engagement-changeset-aware.e2e-spec.ts#L222~L270) is failed in updating properties in which BaseNodes are created in normal situations.

Reply:
Nvm, figure it out.